mpi_learn/mpi/process.py

```diff
@@ -83,10 +83,12 @@ def __init__(self, parent_comm, process_comm, parent_rank=None, num_epochs=1, da
         if self.process_comm is not None and self.process_comm.Get_size() > 1:
             import horovod.common as hvd
             hvd.init(comm=self.process_comm)
-            # After PyTorch integration whis must be updated
-            import horovod.keras as hvdk
             self.algo.worker_optimizer_builder.horovod_wrapper = True
-            #self.callbacks_list.append(hvd.callbacks.BroadcastGlobalVariablesCallback(0))
+            if isinstance(self.model_builder, ModelPytorch):
+                import torch
+                import horovod.torch as hvd
+                torch.cuda.set_device(hvd.local_rank()) # should be properly set with get_device
```

Comment:
@thongonary : if CUDA_VISIBLE_DEVICES="3", should set_device(0) or set_device(3) be used ?
@vloncar can that set_device be put all the way down to build_torch(...) ?

Reply:
or all the way up after get_device, where we know if args.torch is true of false. It might be that this is not necessary, depending on whether set_device has to be done at all

Reply:
we can remove the set_device as this is already implicit